LibraryPanels: Adds folder name to Library Panel card

hugohaggmark opened the pull request:
**What this PR does / why we need it**:
Adds the folder name to the Library Panel card and api:s
![image](https://user-images.githubusercontent.com/562238/117012635-7250ef00-acef-11eb-9e37-045d3834bc13.png)

I changed things on PanelTypeCard so this might need more polish.

**Which issue(s) this PR fixes**:
Fixes #33530

**Special notes for your reviewer**:

hugohaggmark committed:
LibraryPanels: Adds folder filter

hugohaggmark committed:
Refactor: Adds folder filter to library search

hugohaggmark committed:
Refactor: splits huge function into smaller functions

hugohaggmark committed:
LibraryPanels: Adds Panels Page to Manage Folder tabs (#33618)

hugohaggmark committed:
Chore: adds tests to LibraryPanelsSearch

hugohaggmark committed:
Refactor: Adds reducer and tests

hugohaggmark committed:
Merge branch 'master' into hugoh/libpanels-folder

hugohaggmark committed:
Chore: changes GrafanaThemeV2

hugohaggmark committed:
Refactor: adds folderName to get all result

hugohaggmark committed:
Refactor: adds folderName to get result

hugohaggmark committed:
Merge branch 'master' into hugoh/libpanels-foldername

hugohaggmark committed:
Refactor: adds folder name to LibraryPanelDTOMeta

hugohaggmark committed:
Refactor: adds folder name to lbirary panels result

hugohaggmark committed:
Chore: reverts public/app/routes/routes.tsx to master

hugohaggmark commented:
How can we make the folder name clickable? Any ideas?

torkelo commented:
> How can we make the folder name clickable? Any ideas?

should be able to change it into anchor/link , but would need folderUid in the meta

torkelo committed:
Minor style tweak

torkelo commented:
@hugohaggmark pushed a super minor style tweak to make the paddings more consistent, updated folder icon to be small and more inlined with folder name

hugohaggmark commented:
> > How can we make the folder name clickable? Any ideas?
> 
> should be able to change it into anchor/link , but would need folderUid in the meta

How do we solve this for the "General folder" @torkelo? You can't navigate to that right?

hugohaggmark committed:
Refactor: adds folder uid to meta

hugohaggmark commented:
Added folder uid to meta and fixed link and changed icon if there's a UID and different icon when missing the uid, see General folder in picture below:
![image](https://user-images.githubusercontent.com/562238/117105276-f6ea4e80-ad7d-11eb-814a-8eda12fdc0de.png)

dprokop commented on public/app/features/library-panels/components/LibraryPanelCard/LibraryPanelCard.tsx:
```diff
@@ -48,3 +52,50 @@ export const LibraryPanelCard: React.FC<LibraryPanelCardProps & { children?: JSX
+  }
+
+  return (
+    <a href="#" onClick={onNavigateToFolder}>
```
Think you can use `Link` component from react-router-dom to avoid the onNavigateToFolder callback

hugohaggmark replied:
Aha nice, didn't know about that. Thank you!

torkelo replied:
why is not a normal url used here and instead onClick used?  the url should on the anchor should take precedence over the card click I think

hugohaggmark replied:
You're probably right Torkel, I'll try it out.

hugohaggmark replied:
Fixed

hugohaggmark committed:
Chore: updates after PR comments
